cpp/perspective/src/cpp/flat_traversal.cpp

```diff
@@ -243,25 +243,37 @@ t_ftrav::step_begin() {
 
 void
 t_ftrav::step_end() {
+    // The new number of rows in this traversal
     t_index new_size = m_index->size() + m_step_inserts - m_step_deletes;
 
     auto new_index = std::make_shared<std::vector<t_mselem>>();
     new_index->reserve(new_size);
 
     t_uindex i = 0;
     t_multisorter sorter(get_sort_orders(m_sortby));
+
     std::vector<t_mselem> new_rows;
     new_rows.reserve(m_new_elems.size());
 
-    for (t_pkmselem_map::const_iterator pkelem_iter = m_new_elems.begin();
-         pkelem_iter != m_new_elems.end(); ++pkelem_iter) {
+    for (
+        tsl::hopscotch_map<t_tscalar, t_mselem>::const_iterator pkelem_iter = m_new_elems.begin();
+        pkelem_iter != m_new_elems.end();
+        ++pkelem_iter) {
         new_rows.push_back(pkelem_iter->second);
     }
+
+    // TODO: int/float/date/datetime pkeys are already sorted here, so if
+    // there was a way to assert that `psp_pkey` is a string typed column,
+    // we can conditional the sort on whether m_sortby.size() > 0 or if
+    // psp_pkey is a string column.
```

Comment:
I think this still needs to be re-sorted - this `std::sort` just guarantees overlapping indices will be contiguous.